src/lib-components/colors.ts

```diff
@@ -36,26 +46,42 @@ export function hslToHex(hParam: number, sParam: number, lParam: number): string
     g = hue2rgb(p, q, h);
     b = hue2rgb(p, q, h - 1 / 3);
   }
-  const toHex = (x: number): string => {
-    const hex = Math.round(x * 255).toString(16);
-    return hex.length === 1 ? `0${hex}` : hex;
-  };
   return `#${toHex(r)}${toHex(g)}${toHex(b)}`;
 }
 
+export function cubeHelix(h: number, s: number, l: number) {
+  const A = -0.14861;
+  const B = +1.78277;
+  const C = -0.29227;
+  const D = -0.90649;
+  const E = +1.97294;
+  const hRad = (h + 120) * (Math.PI / 180);
+  const a = s * l * (1 - l);
+  const cosh = Math.cos(hRad);
+  const sinh = Math.sin(hRad);
+
+  const r = (l + a * (A * cosh + B * sinh));
+  const g = (l + a * (C * cosh + D * sinh));
+  const b = (l + a * (E * cosh));
+
+  return [r, g, b];
+}
+
 /**
  * Generate HSL color from complex number
  * https://github.com/stared/quantum-game/blob/master/js/transition_heatmap.js
  */
 export function colorComplex(re: number, im: number): string {
   const angleInDegrees = ((Math.atan2(im, re) * 360) / TAU + 360) % 360;
-  const r = Math.sqrt(re * re + im * im); // for pure color it should be always 1
-  return hslToHex(angleInDegrees, 100, 100 - 50 * r);
+  const radius = Math.sqrt(re * re + im * im); // for pure color it should be always 1
+  const [r, g, b] = cubeHelix(angleInDegrees, 1, 1 - 0.5 * radius);
+  return `#${toHex(r)}${toHex(g)}${toHex(b)}`;
 }
```

Comment:
Here I would expect *a few* functions, or maybe `complexColors` with a parameter. To add new color schemes and support multiple, not to *change* the previous one.

Reply:
I don't think there is much to abstract here. Each color space convertion function will have its own distinct inputs anyway, and you will need to convert complex number into them in some way. That mapping depends on specific function. Making this more generic doesn't really save you anything - you need both parts to be color space specific. The only "generic" thing left is text formatting, which can be separate.

I guess taking arg/norm of complex should also be a function call, but it is still just an implementation detail. You might not need them in some situations.

I think a better idea would be to have a standardized "complex to color" function signature that takes `Complex` and returns `[r,g,b]`. That way we can substitute those functions to choose any palette you want, or even allow users to implement their own.

Reply:
@Frizi The issue is different - the old one was not kept, but replaced. 
We want to be able to write a few functions.